server/src/main/java/com/linecorp/centraldogma/server/ZooKeeperServerConfig.java

```diff
@@ -55,6 +64,8 @@ public ZooKeeperAddress(@JsonProperty(value = "host", required = true) String ho
         checkArgument(clientPort >= 0 && clientPort <= 65535,
                       "clientPort: %s (expected: 0-65535)", clientPort);
         this.clientPort = clientPort;
+        this.groupId = groupId;
+        this.weight = weight == null ? 1 : weight;
```

Comment:
Just wondering this could be a negative value. If so we should validate. 🤔

Reply:
There is no specification or documentation for negative weights.
Base on the implementation of QuorumHierarchical: 
- The negative server weight is also added to group weight.
- If a group weight is zero, the group falls out of the hierarchical quorums.
https://github.com/apache/zookeeper/blob/4a2d58219b7435c3b8cdf8f7ab04b158c1900223/zookeeper-server/src/main/java/org/apache/zookeeper/server/quorum/flexible/QuorumHierarchical.java#L312-L321

So I think the weight of servers does not need to be validated and it looks good to follow the ZooKeeper policy.

Reply:
In the following example, Group 2 is moved out from hierarchical quorums.
```
group.1=1:2:3
group.2=4:5:6
group.3=7:8:9

weight.1=1
weight.2=1
weight.3=1
weight.4=-2
weight.5=1
weight.6=1
weight.7=1
weight.8=1
weight.9=1
```

Reply:
What if `weight.4=-3`?
It seems like they don't do `weight <= 0` but `weight == 0`.
Does it mean that weight cannot be negative?

Reply:
> What if weight.4=-3?

Good question. I don't think about that. Let me test it how it works.

Reply:
From the documentation, the negative weight seems to be excluded from quorums.
https://zookeeper.apache.org/doc/r3.5.8/zookeeperHierarchicalQuorums.html

> When running the system, we are able to form a quorum once we have a majority of votes from a majority of non-zero-weight groups. Groups that have zero weight are discarded and not considered when forming quorums.

But it is worth to add tests for verification. Let me add more tests for various weights of servers.

Reply:
There is undefined behavior on negative weight in ZooKeeper hierarchical quorums. It seems to be better not to accept negative weights. Now, an `IllegalArgumentException` is raised on negative weights.